core/src/main/java/com/linecorp/armeria/server/file/FileSystemHttpVfs.java

```diff
@@ -43,49 +43,34 @@
     }
 
     @Override
-    public Entry get(String path, @Nullable String contentEncoding) {
+    public HttpFile get(String path, Clock clock,
+                        @Nullable MediaType contentType, @Nullable String contentEncoding) {
         // Replace '/' with the platform dependent file separator if necessary.
         if (FILE_SEPARATOR_IS_NOT_SLASH) {
             path = path.replace(File.separatorChar, '/');
         }
 
-        final File f = new File(rootDir + path);
-        if (!f.isFile() || !f.canRead()) {
-            return Entry.NONE;
-        }
+        final HttpFileBuilder builder = HttpFileBuilder.of(Paths.get(rootDir + path));
+        return build(builder, clock, contentType, contentEncoding);
+    }
 
-        return new FileSystemEntry(f, path, contentEncoding);
+    static HttpFile build(HttpFileBuilder builder,
+                          Clock clock,
+                          @Nullable MediaType contentType,
+                          @Nullable String contentEncoding) {
+        builder.autoDetectedContentType(false);
```

Comment:
nit: I'm curious if we can turn on this feature only when the specified `contentType` and `contentEncoding` are `null`?

Reply:
That's a good idea. I realized that `contentType` is always auto-detected anyway, so I simplified `HttpVfs.get()` method signature a little bit.